packages/legend-application-studio/src/components/editor/edit-panel/mapping-editor/MappingTestEditor.tsx

```diff
@@ -103,8 +106,11 @@ const MappingTestQueryEditor = observer(
     const applicationStore = useApplicationStore();
 
     // actions
-    const editWithQueryBuilder = applicationStore.guardUnhandledError(
-      async () => {
+    const editWithQueryBuilder = (
+      openQueryBuilderTextEditor: boolean,
```

Comment:
I think it's a bit confusing to have two different flags here. I understand why you did that. Can you please add a note explaining that